### PR TITLE
Domd and mmc integration

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**

### CODE REVIEW

File: boards/rcar_spider.conf [hunk 1]
```diff
@@ -11,3 +11,25 @@ CONFIG_MMC_RCAR=y
...
+CONFIG_SHELL_BACKEND_SERIAL=y
+
+# Enable xen domain support
+
+CONFIG_XEN_INITIAL_DOMAIN=y
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_DOMAIN_MANAGEMENT=y
+CONFIG_XEN_LIBFDT=y
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_SHELL=y
```
Comment (al1img) [inline boards/rcar_spider.conf:35]: New line at the end

File: .clang-format [hunk 2]
```diff
@@ -5,9 +5,6 @@ NamespaceIndentation: None
...
 AlwaysBreakTemplateDeclarations: Yes
-AlignConsecutiveMacros:
-  Enabled: true
-  AcrossEmptyLines: false
-AlignConsecutiveDeclarations:
-  Enabled: true
-  AcrossEmptyLines: false
+AlignConsecutiveMacros: true
+AlignConsecutiveDeclarations: true
+
```
Comment (al1img) [inline .clang-format:10]: Remove empty line

File: west.yml [hunk 3]
```diff
@@ -24,10 +24,13 @@ manifest:
...
 
+    - name: dsemenets
+      url-base: https://github.com/dsemenets
+
   projects:
     - name: zephyr
-      remote: xen-troops
-      revision: aos-wip
+      remote: dsemenets
+      revision: spider_board
```
Comment (xakep-amatop) [inline west.yml:33]: (question) maybe is it better to wait until I merge related changes to aos-wip?

Comment (oleksiimoisieiev) [reply west.yml:33]: I know. But we have no time ((

File: src/main.cpp [hunk 4]
```diff
@@ -15,11 +15,26 @@
 #include "logger/logger.hpp"
 #include "version.hpp"
 
+#if defined(CONFIG_BOARD_RCAR_SPIDER) || defined(CONFIG_BOARD_RCAR_SALVATOR_XS_M3_2X4G)
```
Comment (xakep-amatop) [inline src/main.cpp:18]: (note) you can use `CONFIG_SOC_SERIES_RCAR_GEN4` and `CONFIG_SOC_SERIES_RCAR_GEN3` instead
if we want to add other boards in the future

File: src/domd_cfg_spider.c [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+ */
+
+#include <domain.h>
+#include <string.h>
+#include <zephyr/xen/public/domctl.h>
+
+static char* domd_dtdevs[] = {
+    "/soc/dma-controller@e7350000",
+    "/soc/dma-controller@e7351000",
+    "/soc/mmc@ee140000",
```
Comment (xakep-amatop) [inline src/domd_cfg_spider.c:14]: (note) I'm not sure that this is correct, at far as I know we should delete xen pass through property for MMC and I thought that domd doesn't able to work with this device after it

Comment (oleksiimoisieiev) [reply src/domd_cfg_spider.c]: dropped mmc node from the config

File: src/domd_cfg_gen3.c [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+    "/soc/dma-controller@ec700000",
+    "/soc/dma-controller@ec720000",
+    "/soc/dma-controller@e65a0000",
+    "/soc/dma-controller@e65b0000",
+    "/soc/usb@ee000000",
+    "/soc/usb@ee080000",
+    "/soc/usb@ee080100",
+    "/soc/usb@ee0a0000",
+    "/soc/usb@ee0a0100",
+    "/soc/mmc@ee100000",
```
[the remark this answers is not in the document]
Comment (oleksiimoisieiev) [reply src/domd_cfg_gen3.c:25]: +

File: boards/rcar_h3ulcb_ca57.conf [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+# Enable xen domain support
+
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_DOMAIN_MANAGEMENT=y
+CONFIG_XEN_LIBFDT=y
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_SHELL=y
+
+CONFIG_XRUN=y
```
Comment (xakep-amatop) [inline boards/rcar_h3ulcb_ca57.conf:32]: (note) there isn't new line at the end of the file

**Review by xakep-amatop — APPROVED** [review]
review has been finished, I haven't found major issues
pls, run checkpath and fix all code style issues

`Acked-by: Mykola Kvach <mykola_kvach@epam.com>`

File: boards/rcar_salvator_xs_m3_2x4g.conf [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+# Enable xen domain support
+
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_DOMAIN_MANAGEMENT=y
+CONFIG_XEN_LIBFDT=y
+CONFIG_XEN_STORE_SRV=y
+CONFIG_XEN_SHELL=y
+
+CONFIG_XRUN=y
```
[the remark this answers is not in the document]
Comment (oleksiimoisieiev) [reply boards/rcar_salvator_xs_m3_2x4g.conf]: +

**Pull request merged**